plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -199,28 +217,34 @@ public void onAfter() {
         return new DeleteResult(blobsDeleted.get(), bytesDeleted.get());
     }
 
+    private Iterable<ListBlobItem> getListBlobItems(CloudBlobContainer blobContainer, String path, OperationContext opContext) {
```

Comment:
NIT: let's inline this in `deleteBlobDirectory`

```java
   for (final ListBlobItem blobItem :
         blobContainer.listBlobs(path, true, EnumSet.noneOf(BlobListingDetails.class), null, context)) {
```

where it's used. It's weird having an extra method here when we don't have one for the directory listing in `#children()`.